puppeteer-tests/src/comments/collaboration-test.spec.tsx

```diff
@@ -0,0 +1,71 @@
+import type { Page } from 'puppeteer'
+import { setupBrowser, wait } from '../utils'
+
+const TIMEOUT = 120000
+
+const BRANCH_NAME = process.env.BRANCH_NAME ? `&branch_name=${process.env.BRANCH_NAME}` : ''
+
+async function signIn(page: Page) {
+  const signInButton = await page.waitForSelector('div[data-testid="sign-in-button"]')
+  await signInButton!.click()
+  await page.waitForSelector('#playground-scene') // wait for the scene to render
+}
+
+async function clickCanvasContainer(page: Page, { x, y }: { x: number; y: number }) {
+  const canvasControlsContainer = await page.waitForSelector('#new-canvas-controls-container')
+  await canvasControlsContainer!.click({ offset: { x, y } })
+}
+
+async function expectNSelectors(page: Page, selector: string, n: number) {
+  const elementsMatchingSelector = await page.$$(selector)
+  expect(elementsMatchingSelector).toHaveLength(n)
+}
+
+describe('Comments test', () => {
+  it(
+    'can place a comment',
+    async () => {
+      const [{ page: page1, browser: browser1 }, { page: page2, browser: browser2 }] =
```

Comment:
Not a huge issue but I'd find this easier to follow if it looked like this:
```typescript
const browser1Promise = setupBrowser(...)
const browser2Promise = setupBrowser(...)
const {page: page1, browser: browser1} = await browser1Promise
const {page: page2, browser: browser2} = await browser2Promise
```

Reply:
good idea, implemented in https://github.com/concrete-utopia/utopia/pull/4702/commits/14a36f563e78c920b24a7a0fa7aeca2250789183